src/JSArrayProxy.cc

```diff
@@ -1130,26 +1232,116 @@ PyObject *JSArrayProxyMethodDefinitions::JSArrayProxy_sort(JSArrayProxy *self, P
   if (JSArrayProxy_length(self) > 1) {
     JS::RootedValue jReturnedArray(GLOBAL_CX);
     if (keyfunc != Py_None) {
-      if (!PyFunction_Check(keyfunc) && !PyCFunction_Check(keyfunc)) {
-        PyErr_Format(PyExc_TypeError, "'%.200s' object is not callable", Py_TYPE(keyfunc)->tp_name);
-        return NULL;
+      if (PyFunction_Check(keyfunc)) {
+        // we got a python key function, check if two-argument js style or standard python 1-arg
+        PyObject *code = PyFunction_GetCode(keyfunc);
+        if (((PyCodeObject *)code)->co_argcount == 1) {
+          // adapt to python style, provide js-style comp wrapper that does it the python way, which is < based with calls to keyFunc
+          JS::RootedObject funObj(GLOBAL_CX, JS_GetFunctionObject(JS_NewFunction(GLOBAL_CX, sort_compare_key_func, 2, 0, NULL)));
+
+          JS::RootedValue privateValue(GLOBAL_CX, JS::PrivateValue(keyfunc));
+          if (!JS_SetProperty(GLOBAL_CX, funObj, "_key_func_param", privateValue)) {  // JS::SetReservedSlot(functionObj, KeyFuncSlot, JS::PrivateValue(keyfunc)); does not work
+            PyErr_Format(PyExc_SystemError, "%s JSAPI call failed", JSArrayProxyType.tp_name);
+            return NULL;
+          }
+
+          JS::RootedValue reverseValue(GLOBAL_CX);
+          reverseValue.setBoolean(reverse);
+          if (!JS_SetProperty(GLOBAL_CX, funObj, "_reverse_param", reverseValue)) {
+            PyErr_Format(PyExc_SystemError, "%s JSAPI call failed", JSArrayProxyType.tp_name);
+            return NULL;
+          }
```

Comment:
Why does this not work?:
`js::SetFunctionNativeReserved(funObj, 0, JS::PrivateValue((void *)keyfunc));`

To get the private slot value back in sort_compare_key_func, you should be able to do:
```cpp
PyObject *keyfunc = (PyObject*)(js::GetFunctionNativeReserved(&(callargs.callee()), 0).toPrivate());

Reply:
Still does not work, the internals of SetReservedSlot don't like this type of object, we get 0 fixedSlots